lib/matplotlib/axes/_base.py

```diff
@@ -4335,8 +4335,7 @@ def get_tightbbox(self, renderer, call_axes_locator=True,
         for a in bbox_artists:
             bbox = a.get_tightbbox(renderer)
             if (bbox is not None and
-                    (bbox.width != 0 or bbox.height != 0) and
-                    np.isfinite(bbox.width) and np.isfinite(bbox.height)):
+                    0 < bbox.width < np.inf and 0 < bbox.height < np.inf):
```

Comment:
Optional, but might be easier to read when wrapping the height check to a separate line.